src/Xamarin.CodeAnalysis/RequiresSuperAttributeAnalyzer.cs

```diff
@@ -0,0 +1,106 @@
+			"Notifications",
+			DiagnosticSeverity.Info,
+			isEnabledByDefault: true
+			);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create (Rule);
+
+		const string RequiresSuperAttributeName = "RequiresSuperAttribute";
+		const string RequiresSuperAttributeNamespace = "ObjCRuntime";
+
+		public override void Initialize (AnalysisContext context)
+		{
+			context.ConfigureGeneratedCodeAnalysis (GeneratedCodeAnalysisFlags.None);
+			context.EnableConcurrentExecution ();
+			context.RegisterOperationBlockAction ((codeBlockContext) => {
+				// returns if the code block isn't an overridden method
+				var method = (IMethodSymbol)codeBlockContext.OwningSymbol;
+				if (method == null || !method.IsOverride) { return; }
+
+				// seeing if the [RequiresSuper] attribute is there
+				var baseType = method.ContainingType.BaseType;
+				ISymbol baseMethod = baseType.GetMembers (method.Name).FirstOrDefault ();
+				// returns if the [RequiresSuper] attribute isn't present
+				if (!baseMethod.GetAttributes ().Any (attr => (attr.AttributeClass.Name == RequiresSuperAttributeName && attr.AttributeClass.ContainingNamespace.Name == RequiresSuperAttributeNamespace))) {
+					return;
+				}
+
+				foreach (var block in codeBlockContext.OperationBlocks.Where ((b) => b.Kind == OperationKind.Block)) // loops once per method
+				{
+					foreach (var exp in block.Children) // loops once per operation in the method body 
+					{
+						//skips if the operation isn't base.___ or return ___; 
+						if (exp.Kind != OperationKind.ExpressionStatement && exp.Kind != OperationKind.Return) { continue; }
+
+
+						foreach (var child in exp.Children) {
+							// skips if it's not an Invocation Operation
+							if (child.Kind != OperationKind.Invocation) { continue; }
+							var invocationOp = (IInvocationOperation)child;
+							var instance = invocationOp.Instance;
+							if (instance == null) { return; } // FIXME: this is the void foo () => . . .case; make this register a diagnostic (the code fix currently throws an error if the diagnostic is registered)
+
+							// skips if the operation doesn't contain base
+							if (instance.Kind != OperationKind.InstanceReference) { continue; }
+							var instanceOp = (IInstanceReferenceOperation)instance;
+
+							// checks if in the base.[method]() call, [method] matches the one in the method's signature
+							if (instanceOp.ReferenceKind != InstanceReferenceKind.ContainingTypeInstance) { continue; }
+
+							var invokedMethod = invocationOp.TargetMethod;
+							var containingMethod = (IMethodSymbol)codeBlockContext.OwningSymbol;
+
+							var iter = containingMethod;
+							while (iter != null) {
+								iter = iter.OverriddenMethod;
+								if (invokedMethod.Equals (iter)) {
+									return;
+								}
+							}
+						}
+					}
+				}
+				//adds the diagnostic if we don't find base.method() or return base.method()
```

Comment:
Great job putting super helpful comments throughout!

Reply:
Given chamons' comment [here](https://github.com/xamarin/CodeAnalysis/pull/2#issuecomment-579969501), I'll mark this comment as resolved for now (and will work on trying to get the testing framework set up with @sharwell; if I don't end up completing it before I leave, then they will be filled in later.